### PR TITLE
media-sound/denemo: fix build for gcc-10

### PR DESCRIPTION
**Pull request by waebbl**
- add patches to work with gcc-10 -fno-common flag
- install appdata/metainfo into new recommended location

Closes: https://bugs.gentoo.org/712926
Package-Manager: Portage-2.3.94, Repoman-2.3.21
Signed-off-by: Bernd Waibel <waebbl@gmail.com>

### CODE REVIEW

**Comment by a17r** [comment]
let's drop `r0` in a second commit.

File: media-sound/denemo/denemo-2.3.0-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	evince? ( >=app-text/evince-3.22.1-r1:= )
+	fluidsynth? ( >=media-sound/fluidsynth-1.1.6-r1:= )
+	gtk3? (
+		x11-libs/gtk+:3
+		x11-libs/gtksourceview:3.0=
+	)
+	!gtk3? (
+		x11-libs/gtk+:2
+		x11-libs/gtksourceview:2.0
+	)
```
Comment (a17r) [inline media-sound/denemo/denemo-2.3.0-r1.ebuild]: Would it be a good opportunity to get rid of GTK2 support here?

Comment (waebbl) [reply media-sound/denemo/denemo-2.3.0-r1.ebuild]: Good point. It should work, I'm going to check this.

Comment (waebbl) [reply media-sound/denemo/denemo-2.3.0-r1.ebuild]: Had to remove gtk3 and evince options to work it out.

File: media-sound/denemo/denemo-2.3.0-r1.ebuild [hunk 2]
```diff
@@ -17,31 +17,25 @@ KEYWORDS="~amd64 ~x86"
 # --enable-mem(no) memory debugging: needs Electric fence (efence), which
 #		is not available in portage. See https://github.com/boundarydevices/efence
 # --enable-gtk-doc-pdf(no) doesn't work
-IUSE="alsa +aubio debug +evince jack +fluidsynth gtk-doc +gtk3 nls +portaudio
-	+portmidi +rubberband static test"
+IUSE="alsa +aubio debug jack +fluidsynth gtk-doc nls +portaudio +portmidi
+	+rubberband static test"
```
Comment (a17r) [inline media-sound/denemo/denemo-2.3.0-r1.ebuild]: shouldn't it be `static-libs` instead? what does it do exactly?

Comment (waebbl) [reply media-sound/denemo/denemo-2.3.0-r1.ebuild]: Good catch. Looks like it's a no-op. Configure has an --enable-static option, where the comment says it's to build static libraries, but the app doesn't install any libraries. I guess, we should remove it.

Comment (waebbl) [reply media-sound/denemo/denemo-2.3.0-r1.ebuild]: The binaries are still dynamically linked even with `USE=static`

Comment (a17r) [reply media-sound/denemo/denemo-2.3.0-r1.ebuild]: yeah, I say drop it. if no one asks for it, and a package has no usedeps, drop static-libs wherever you encounter it.

Comment (waebbl) [reply media-sound/denemo/denemo-2.3.0-r1.ebuild]: alright

**Comment by a17r** [comment]
Interestingly wasn't able to build 3.4.0-r0 with USE="-aubio -portaudio -portmidi -rubberband -evince":
```
/usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: scripting/scheme-callbacks.o: in function `scheme_PrintHistory':
scheme-callbacks.c:(.text+0x2653): undefined reference to `display_printhistory'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: scripting/scheme-callbacks.o: in function `scheme_swap_display_and_source':
scheme-callbacks.c:(.text+0x3ff4): undefined reference to `get_source_position'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: scheme-callbacks.c:(.text+0x4052): undefined reference to `move_source_window'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: core/menusystem.o:(.data.rel.ro+0x2ed8): undefined reference to `display_printhistory'
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:1154: denemo] Error 1
```

**Comment by waebbl** [comment]
I had the same issue, when I tested for removing gtk2. It was the reason, why I removed evince USE flag and hard depend on it.
I'm not sure, what causes this. Is it a somewhat unclean implementation of HAVE_EVINCE by upstream, or is it caused by inter-dependencies of packages caused by USE flags on Gentoo?

**Comment by a17r** [comment]
could be a simple ifdef issue in upstream's build system.

**Comment by waebbl** [comment]
Could be, there are a few files, which #include <evince-[view,document].h> unconditionally but are not included in the sources list for HAVE_EVINCE.

**Pull request closed without merging**